source/GameWindow.cpp

```diff
@@ -82,7 +82,11 @@ bool GameWindow::Init(bool headless)
 
 	// When running the integration tests, don't create a window nor an OpenGL context.
 	if(headless)
+#if defined(__linux__) && !SDL_VERSION_ATLEAST(2, 0, 22)
+		setenv("SDL_VIDEODRIVER", "dummy");
```

Comment:
This fails to compile because it is missing the mandatory third argument to setenv: `overwrite`.

```suggestion
		setenv("SDL_VIDEODRIVER", "dummy", 0);
```

That argument tells what to do if the variable is already set. It should be 0, indicating that the variable should not be replaced if it is already set. This allows the caller to force the integration tests to run with a head.

Reply:
It actually should be replaced, it doesn't make sense to show the window when nothing gets drawn to the screen anyways.